### ISSUE TITLE
Use of MCP23017 I2C version

### ISSUE THREAD

**Issue by IraSch** [taconsultingllc/hornet-nest-alarm-panel#1]
It is my understanding (perhaps wrong) that the MCP230xx family with I2C interface suffers from a rare but still present bug where a change a input at just the wrong time can lock up the I2C bus.  Given that the alarm panel will be operating for years at a time,  and that an input change that meets the "bug" criteria would freeze the panel, wouldn't it be more prudent to use the SPI version of the the chip that doesn't suffer from the bug?

**Comment by taconsultingllc** [comment]
You're not wrong and it's been well documented as to the cause of the issue.  In fact, MicroChip revised their documentation in 2022 for the chip, derating it from a 16 I/O mux to a 14 I/O + 2 mux.  GPA7 and GPB7 have been identified as the culprit and are now only intended to be as outputs.  I avoided these pins for the zone inputs but do have plans in the next revision to expose these 6 pins as logic level outputs on the board.

https://hackaday.com/2023/02/03/mcp23017-went-through-shortage-hell-lost-two-inputs/

**Comment by IraSch** [comment]
I thought what I read implied that any input line changing at the wrong time could corrupt the I2C bus, but you may be correct in that it’s only related to bit 7. Hopefully that’s true. 

 

Ira
